src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+        }
+
+        if (!data.get(StorageKeys.SEARCH_OFFSET)) {
+          this.core.storage.set(StorageKeys.SEARCH_OFFSET, 0);
+        }
+
+        data.forEach((value, key) => {
+          if (key === StorageKeys.QUERY) {
+            return;
+          }
+          const parsedValue = this._parsePersistentStorageValue(key, value);
+          this.core.storage.set(key, parsedValue);
+        });
+
+        this.core.storage.set(StorageKeys.HISTORY_POP_STATE, data);
+
+        if (hasQuery) {
+          this.core.storage.set(StorageKeys.QUERY, query);
+        }
+      },
+      persistedValueParser: this._parsePersistentStorageValue
+    });
+    storage.init(window.location.search);
+    storage.set(StorageKeys.SEARCH_CONFIG, parsedConfig.search);
+    storage.set(StorageKeys.VERTICAL_PAGES_CONFIG, parsedConfig.verticalPages);
+    storage.set(StorageKeys.LOCALE, parsedConfig.locale);
+    storage.set(StorageKeys.QUERY_SOURCE, parsedConfig.querySource);
+
+    // Check if sessionsOptIn data is stored in the URL. If it is, prefer that over
+    // what is in parsedConfig.
+    const sessionOptIn = storage.get(StorageKeys.SESSIONS_OPT_IN);
+    if (!sessionOptIn) {
+      storage.set(
+        StorageKeys.SESSIONS_OPT_IN,
+        { value: parsedConfig.sessionTrackingEnabled, setDynamically: false });
+    } else {
+      // If sessionsOptIn was stored in the URL, it was stored only as a string.
+      // Parse this value and add it back to storage.
+      storage.set(
+        StorageKeys.SESSIONS_OPT_IN,
+        { value: (/^true$/i).test(sessionOptIn), setDynamically: true });
+    }
+
+    parsedConfig.noResults && storage.set(StorageKeys.NO_RESULTS_CONFIG, parsedConfig.noResults);
+    const isSuggestQueryTrigger =
+      storage.get(StorageKeys.QUERY_TRIGGER) === QueryTriggers.SUGGEST;
+    if (storage.has(StorageKeys.QUERY) && !isSuggestQueryTrigger) {
+      storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.QUERY_PARAMETER);
+    }
+
+    const context = storage.get(StorageKeys.API_CONTEXT);
+    if (context && !isValidContext(context)) {
+      storage.delete(StorageKeys.API_CONTEXT);
+      console.error(`Context parameter "${context}" is invalid, omitting from the search.`);
+    }
+
+    if (storage.get(StorageKeys.REFERRER_PAGE_URL) === undefined) {
+      storage.set(
+        StorageKeys.REFERRER_PAGE_URL,
+        urlWithoutQueryParamsAndHash(document.referrer)
+      );
+    }
+
+    this._services = getServices(parsedConfig, storage);
+
+    this._eligibleForAnalytics = parsedConfig.businessId != null;
+    // TODO(amullings): Initialize with other services
+    if (this._eligibleForAnalytics) {
+      this._analyticsReporterService = new AnalyticsReporter(
+        parsedConfig.experienceKey,
+        parsedConfig.experienceVersion,
+        parsedConfig.businessId,
+        parsedConfig.analyticsOptions,
+        parsedConfig.environment);
+
+      // listen to query id updates
+      storage.registerListener({
```

Comment:
does the redirect searchbar ever get a query id?

Reply:
Since a search is never actually fired on a redirect searchbar, I don't think there will ever be a query id. So, I will be removing this listener.